stable/mysql/templates/deployment.yaml

```diff
@@ -3,10 +3,15 @@ kind: Deployment
 metadata:
   name: {{ template "mysql.fullname" . }}
   labels:
-    app: {{ template "mysql.fullname" . }}
     chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
     release: "{{ .Release.Name }}"
     heritage: "{{ .Release.Service }}"
+    {{- if not .Values.deploymentLabels.app }}
```

Comment:
now i realize this, i think this is will not work properly.
did you test that?

Reply:
why remove the app label?

Reply:
Yes, I think this will work properly and I've tested it.

If you did not set `app` in `deploymentLabels`, it will be set as it was before (with `template "mysql.fullname"`. However if it is set, that value will be used.

Reply:
The app label is used by the svc selector using the fullname helper, so overriding it in with `deploymentLabels` will break.

I also find that functionality confusing at best and one can already override that particular label in a safe way using `fullnameOverride`

Reply:
But `fullnameOverride` does not always make sense.

Suppose the following scenario: You have a project consisting of two Deployments. One is the Webserver and the other one is the db (mysql in that case). In OpenShift I use the label app to group them together by its project name. But with the current mysql Helm chart I have  either mysql in the project name, which is wrong or I can not group them together.

With my changes I have the flexibility to do both. The odd syntax to check if label `app` exists is just to ensure backwards compatibility.

Reply:
That is not entirely how the `app` label is meant to be used though. If you have a project it would have one chart depending on the webserver chart and the mysql chart. They would then have the same `release` label. So `release` would be what you group by.